protobuf-ckb-syscalls/src/lib.rs

```diff
@@ -381,3 +383,202 @@ impl SyscallImpls for ProtobufBasedSyscallImpls {
         self.io_syscall(buf, offset, None)
     }
 }
+
+// TODO: it remains a question if this should be it's own type, or we should
```

Comment:
This `TODO` is posed as a question for discussion, I'm not sure if we should split `ProtobufVmRunnerImpls` into a separate type, or have `ProtobufImpls` implement `CkbvmRunnerImpls`.

* Use a single `ProtobufImpls` avoid repetition, but saves extra coding
* Use 2 types clearly distinguish between 2 uses, but we have to implement a dummy `SyscallImpls` on `ProtobufVmRunnerImpls` again.

Reply:
Likewise, I can't say which behavior is more correct. But I tend to let each code do its own thing.

Reply:
In that sense feel free to just remove the TODOs.

Reply:
Done